src-input/duk_hobject_props.c

```diff
@@ -5663,7 +5663,7 @@ duk_bool_t duk_hobject_define_property_helper(duk_context *ctx,
 			a = (duk_harray *) obj;
 			DUK_DD(DUK_DDPRINT("Object.defineProperty() attribute update for duk_harray .length -> %02lx", (unsigned long) new_flags));
 			DUK_ASSERT_HARRAY_VALID(a);
-			if ((new_flags & DUK_PROPDESC_FLAGS_EC) != (curr.flags & DUK_PROPDESC_FLAGS_EC)) {
+			if ((new_flags & DUK_PROPDESC_FLAGS_EC) != ((duk_small_uint_t)curr.flags & DUK_PROPDESC_FLAGS_EC)) {
```

Comment:
I'll have to go back and check.  I compile duktape on a bunch of platforms, and I made this fix a while back because it was triggering a warning.  I'll see if I can find out which platform was causing it.